programs/cleos/main.cpp

```diff
@@ -221,6 +221,45 @@ void add_standard_transaction_options(CLI::App* cmd, string default_permission =
    cmd->add_option("--delay-sec", delaysec, localized("Set the delay_sec seconds, defaults to 0s"));
 }
 
+class signing_keys_option {
+public:
+   signing_keys_option() {}
+   void add_option(CLI::App* cmd) {
+      cmd->add_option("--sign-with", public_key_json, localized("The public key or json array of public keys to sign with"));
+   }
+
+   std::vector<public_key_type> get_keys() {
+      std::vector<public_key_type> signing_keys;
+      if (!public_key_json.empty()) {
+         if (boost::istarts_with(public_key_json, "EOS") || boost::istarts_with(public_key_json, "PUB_R1")) {
```

Comment:
How about adding a static method `is_public_key_str()` that checks for "EOS' and "PUB_R1". Use it here and in `parse_json_authority_or_key`.

Also, shouldn't the valid strings be "EOS", "PUB_R1", "PUB_K1", and "PUB_WA" ? @spoonincode

Reply:
Yes those are the 4 current valid prefixes. Though, keosd can't sign PUB_WA so I'm not sure the usefulness of it being handled; probably still want it for completeness sake